kedro/extras/datasets/spark/spark_dataset.py

```diff
@@ -401,10 +401,12 @@ def _exists(self) -> bool:
         try:
             self._get_spark().read.load(load_path, self._file_format)
         except AnalysisException as exception:
-            if (
-                exception.desc.startswith("Path does not exist:")
-                or "is not a Delta table" in exception.desc
-            ):
+            if hasattr(exception, "desc"):  # Pyspark <3.4
+                message = exception.desc
+            else:
+                message = exception.message
+
+            if "Path does not exist:" in message or "is not a Delta table" in message:
```

Comment:
In theory we should freeze it, but this minor change would allow us release Python 3.11 without breaking change. We should also have this change in `kedro-datasets`